Amato App Submission

ziadgit opened the pull request:
# App Submission

### App name
Amato Timer

### 256x256 SVG icon
_(Submit an icon with no rounded corners as it will be dynamically rounded with CSS. GitHub doesn't allow uploading SVGs directly, so please upload your icon to an alternate service, like https://svgur.com, and paste the link below.)_

https://svgshare.com/i/wdB.svg

### Gallery images
_(Upload 3 to 5 high-quality gallery images (1440x900px) of your app in PNG format, or just upload 3 to 5 screenshots of your app and we'll help you design the gallery images.)_

![1](https://github.com/getumbrel/umbrel-apps/assets/7642/d2eb43f4-2d8f-48a5-b136-cf6d70e052b9)
![2](https://github.com/getumbrel/umbrel-apps/assets/7642/7bdfc9bb-3447-439e-8a4e-7886b71a0932)
![3](https://github.com/getumbrel/umbrel-apps/assets/7642/33eddf3e-36bc-4fe0-a68d-9ec4e3dd6d56)


### I have tested my app on:
- [ ] [Umbrel dev environment](https://github.com/getumbrel/umbrel-dev)
- [x] [Umbrel OS on a Raspberry Pi 4](https://github.com/getumbrel/umbrel-os)
- [ ] [Custom Umbrel install on Linux](https://github.com/getumbrel/umbrel#-installation)

ParthJadhav commented on amato/umbrel-app.yml:
```diff
@@ -0,0 +1,20 @@
+description: >-
+  A simple timer with default 25 minute increments to track your work.
+releaseNotes: >-
+  This is an initial alpha version
```
You can leave this empty as `releaseNotes` are shown for updates.

```suggestion
  releaseNotes: ""
```

ParthJadhav commented:
Also @ziadgit , Can you add direct link to SVG to the description:

https://svgshare.com/i/wdB.svg

ParthJadhav reviewed (changes requested):
Tested on Umbrel-Dev server, Works as expected! 

feel free to make the changes listed above 👆

Thanks for the  contribution  ❤️

ParthJadhav commented on amato/umbrel-app.yml:
```diff
@@ -7,7 +7,7 @@ tagline: Simple work timer.
   A simple timer with default 25 minute increments to track your work.
 releaseNotes: >-
-  This is an initial alpha version
+  releaseNotes: ""
```
Apologies for the confusion in my previous comment. Please replace:
```
releaseNotes: >-
  releaseNotes: ""
```
with:
```
releaseNotes: ""
```

nmfretz commented on amato/umbrel-app.yml:
```diff
@@ -0,0 +1,21 @@
+version: "0.0.1.1"
+tagline: Simple work timer.
+description: >-
+  A simple timer with default 25 minute increments to track your work.
```
It looks like it's not really 25 minute "increments" since there are 3 set options of 5 minutes, 10 minutes, and 25 minutes with no ability to increment higher:
![image](https://github.com/getumbrel/umbrel-apps/assets/85373263/365bab4f-2d73-4fce-8d0e-e651d1a0e2a1)

Instead, should we say something like:
`A simple timer to help track your time spent on tasks and boost your productivity. Choose between 5, 10, and 25 minute timers`

nmfretz commented on amato/umbrel-app.yml:
```diff
@@ -0,0 +1,21 @@
+path: ""
+defaultUsername: ""
+defaultPassword: ""
+submitter: ziadgit
```
If you are the developer of this app, then we should make this field and the `developer` field have the same info so that users know that the developer of the app is also the submitter.

nmfretz commented on amato/umbrel-app.yml:
```diff
@@ -0,0 +1,21 @@
+description: >-
+  A simple timer with default 25 minute increments to track your work.
+releaseNotes: ""
+developer: Bitparasol
+website: http://bitparasol.com
```
Is this your company and website? I don't see anything related to Amato timer? These fields are here to provide additional info on the app to users who may be curious.

We should also add the following two fields:
```yaml
repo: <the-repository-where-the-code-lives-if-public>
support: <link-to-support-site-or-github-issues>
```

If you don't add a repo, then the UI will show your source code being private like this:
![image](https://github.com/getumbrel/umbrel-apps/assets/85373263/0a5f2a38-4ceb-4326-850a-c1d0adff6268)


See Core Lightning app as an example:
https://github.com/getumbrel/umbrel-apps/blob/776587169cba9e345fcbc9e54f82675fd10e9eb2/core-lightning/umbrel-app.yml#L19-L24

nmfretz commented on amato/umbrel-app.yml:
```diff
@@ -0,0 +1,21 @@
+manifestVersion: 1
+id: amato
```
pomodoro or amato?

johnpc replied:
[Pomodoro](https://en.wikipedia.org/wiki/Pomodoro_Technique) is the technique of dedicated focus. Often (for reasons that I don't fully understand), "Pomodoro time" is often signified by a tomato. For example, if you are doing focused work and do not want to be bothered, you might set your Slack status to a tomato emoji.

My guess is "Amato" is part of the brand that this developer wants to have for his pomodoro timer, a reference to the tomato. If he named his app Pomodoro, it would be a lot more competitive from an SEO perspective.

nmfretz commented on amato/umbrel-app.yml:
```diff
@@ -0,0 +1,21 @@
+manifestVersion: 1
+id: amato
+category: productivity
+name: Amato Timer
```
Pomodoro Timer or Amato Timer?

nmfretz commented on amato/umbrel-app.yml:
```diff
@@ -0,0 +1,21 @@
+category: productivity
+name: Amato Timer
+version: "0.0.1.1"
+tagline: Simple work timer.
```
We can remove the `.` from the end of the tagline for style-consistency with the rest of the apps in the app store.

nmfretz commented on amato/umbrel-app.yml:
```diff
@@ -0,0 +1,21 @@
+defaultUsername: ""
+defaultPassword: ""
+submitter: ziadgit
+submission: https://github.com/getumbrel/umbrel-apps/pull/729
```
Please also add the following line to this yaml file:
`dependencies: []`

nmfretz reviewed (commented):
Thanks for submitting Amato Timer @ziadgit! I have tested and left a review below.

Are you the developer of this app? I noticed a couple things that should be clarified:
- You have submitted this app as `Amato Timer`, but I see that the UI calls the app `Pomodoro Timer` and the docker image is tagged as `berkeleyengineer/pomoro`. I think we should try to align some/all of these so that it is clear what the app is and where it is coming from. What are your thoughts on this?
- You have listed Bitparasol as the developer and linked to the bitparasol website. Is this referring to you and your site? I did not see anything related to Amato timer on this site. If you are the developer, then we should make `developer` and `submitter` the same, and link to something that describes Amato Timer or shows the source code.

ziadgit commented:
I believe I have addressed all the branding concerns in the UI and the schema and linked to appropriate bitparasol pages. I've also updated the image to reference the specific sha of the latest image. If after testing this is approved I can update the screenshots.

nmfretz commented:
Excellent, thanks for addressing everything @ziadgit.  We can help convert screenshots into gallery images for you and also help format your icon svg. I'll post them here soon for your approval and then we can send this to the app store!

FYI, I changed the `submitter` to "bitparasol" so that it shows up nicely in the UI. Otherwise it ends up looking like this in the app store.

<img width="416" alt="image" src="https://github.com/getumbrel/umbrel-apps/assets/85373263/3f74ed8f-9290-4da5-ba8c-f3099a88630c">

When someone clicks the "bitparasol" hyperlink next to "Submitted by" it brings them to this PR showing you as submitter.

highghlow commented:
Hello, @nmfretz , what's keeping this PR from merging?

nmfretz commented:
Thanks for pinging me here @highghlow.

@ziadgit, I've just reviewed this again. Is Amato Timer a personal project for learning to code and package an app for umbrelOS? I ask becasue a tomato timer is a common tutorial app for programming. If so, we're hesitant to add this to the official app store because there may not be a large incentive for you (or other developers) to keep maintaining the app.

It is awesome that you're building something for umbrelOS. If you're trying to just start out experimenting with a simple app, then there are a couple of other options that might be better at this early stage in Amato Timer's development:

1) Create a Community App Store: https://github.com/getumbrel/umbrel-community-app-store

    You and other users can add this app store to umbrelOS through the UI. This allows you to distribute and test the app while you continue developing it. It's a good way to gather feedback and improve your app with real user input.
    
    
    <img width="591" alt="image" src="https://github.com/getumbrel/umbrel-apps/assets/85373263/f06be24c-ce97-4bb4-a3ca-ff211e41763c">


2) Use the Portainer app on umbrelOS: https://apps.umbrel.com/app/portainer

    Portainer allows you to run any compose file, so you can very quickly set up and test anything through it.
    
 Let me know if either of those options would work for you.

ziadgit commented:
Hi Nathan,

I'd say the plans long term were a bit more ambitious than that. If there
was a learning motivation, it was less the technical aspect, and more the
approval process I was curious about. Is there a place for a pomodoro app
on the store? If so, I'm happy to design and build toward a feature
set/level of design robustness where umbrel is happy to bring the app to
the official store.

On Sun, Jun 23, 2024 at 10:47 PM Nathan Fretz ***@***.***>
wrote:

> Thanks for pinging me here @highghlow <https://github.com/highghlow>.
>
> @ziadgit <https://github.com/ziadgit>, I've just reviewed this again. Is
> Amato Timer a personal project for learning to code and package an app for
> umbrelOS? I ask becasue a tomato timer is a common tutorial app for
> programming. If so, we're hesitant to add this to the official app store
> because there may not be a large incentive for you (or other developers) to
> keep maintaining the app.
>
> It is awesome that you're building something for umbrelOS. If you're
> trying to just start out experimenting with a simple app, then there are a
> couple of other options that might be better at this early stage in Amato
> Timer's development:
>
>    1.
>
>    Create a Community App Store:
>    https://github.com/getumbrel/umbrel-community-app-store
>
>    You and other users can add this app store to umbrelOS through the UI.
>    This allows you to distribute and test the app while you continue
>    developing it. It's a good way to gather feedback and improve your app with
>    real user input.
>    image.png (view on web)
>    <https://github.com/getumbrel/umbrel-apps/assets/85373263/f06be24c-ce97-4bb4-a3ca-ff211e41763c>
>    2.
>
>    Use the Portainer app on umbrelOS:
>    https://apps.umbrel.com/app/portainer
>
>    Portainer allows you to run any compose file, so you can very quickly
>    set up and test anything through it.
>
> Let me know if either of those options would work for you.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/getumbrel/umbrel-apps/pull/729#issuecomment-2185661060>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAB3WWG5PFJ3CXEVJHKFVTZI6XH7AVCNFSM6AAAAAA3USDCSOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCOBVGY3DCMBWGA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

nmfretz commented:
Understood, thanks for that explanation @ziadgit!

If you're comfortable with this, let's do the following:

I see your app is primarily built with a basic HTML structure and minimal custom styling, relying on the browser’s default rendering of HTML elements like `<h1>` and `<button>`. Would you consider adding custom styling (through vanilla CSS or a component library or framework) so that it can stand the test of time against other Tomato Timer apps out there even if you move on to other things and end up never updating Amato again?

nmfretz commented:
@ziadgit I am going to close this PR for now, but please reopen when/if you end up working on Amato again.

ziadgit commented:
Hi Nmfretz,

I'm working on the changes, but have been moving the bitparasol domain this week.
